client/telemetry/TelemetryEventClient.go

```diff
@@ -0,0 +1,297 @@
+	if err != nil {
+		fmt.Println("error in starting heartbeat event")
+		return nil, err
+	}
+	return watcher, err
+}
+
+func (impl *TelemetryEventClientImpl) StopCron() {
+	impl.cron.Stop()
+}
+
+func (impl *TelemetryEventClientImpl) Watch() {
+	impl.logger.Infow("starting git watch thread")
+	impl.logger.Infow("stop git watch thread")
+}
+
+type TelemetryEventDto struct {
+	UCID           string             `json:"ucid"` //unique client id
+	Timestamp      time.Time          `json:"timestamp"`
+	EventMessage   string             `json:"eventMessage"`
+	EventType      TelemetryEventType `json:"eventType"`
+	Summery        *SummeryDto        `json:"summery"`
+	ServerVersion  string             `json:"serverVersion"`
+	DevtronVersion string             `json:"devtronVersion"`
+}
+
+type SummeryDto struct {
+	ProdAppCount            int `json:"prodAppCount"`
+	NonProdAppCount         int `json:"nonProdAppCount"`
+	UserCount               int `json:"userCount"`
+	EnvironmentCount        int `json:"environmentCount"`
+	ClusterCount            int `json:"nonProdApps"`
+	CiCountPerDay           int `json:"ciCountPerDay"`
+	CdCountPerDay           int `json:"cdCountPerDay"`
+	HelmChartCount          int `json:"helmChartCount"`
+	SecurityScanCountPerDay int `json:"securityScanCountPerDay"`
+}
+
+const DevtronUniqueClientIdConfigMap = "devtron-ucid"
+
+type TelemetryEventType int
+
+const (
+	Heartbeat TelemetryEventType = iota
+	InstallationStart
+	InstallationSuccess
+	InstallationFailure
+	UpgradeSuccess
+	UpgradeFailure
+	Summery
+)
+
+func (d TelemetryEventType) String() string {
+	return [...]string{"Heartbeat", "InstallationStart", "InstallationSuccess", "InstallationFailure", "UpgradeSuccess", "UpgradeFailure", "Summery"}[d]
+}
+
+func (impl *TelemetryEventClientImpl) SummeryEventForTelemetry() {
+	impl.logger.Info(">>>>>>>>>>VIKI startup summery event")
+
+	clusterBean, err := impl.clusterService.FindOne(cluster.ClusterName)
+	if err != nil {
+		return
+	}
+	cfg, err := impl.clusterService.GetClusterConfig(clusterBean)
+	if err != nil {
+		return
+	}
+
+	client, err := impl.K8sUtil.GetClient(cfg)
+	if err != nil {
+		return
+	}
+	cm, err := impl.K8sUtil.GetConfigMapFast(impl.aCDAuthConfig.ACDConfigMapNamespace, DevtronUniqueClientIdConfigMap, client)
+	if err != nil && strings.Contains(err.Error(), "not found") {
+		// if not found, create new cm
+		//cm = &v12.ConfigMap{ObjectMeta: v13.ObjectMeta{Name: "devtron-upid"}}
+		cm = &v1.ConfigMap{ObjectMeta: v12.ObjectMeta{Name: "devtron-upid"}}
+		data := map[string]string{}
+		data["UCID"] = util.Generate(16) // generate unique random number
+		cm.Data = data
+		_, err = impl.K8sUtil.CreateConfigMapFast(impl.aCDAuthConfig.ACDConfigMapNamespace, cm, client)
+		if err != nil {
+			return
+		}
+	}
+	if cm == nil {
+		return
+	}
+	dataMap := cm.Data
+	ucid := dataMap["UCID"]
+
+	discoveryClient, err := impl.K8sUtil.GetK8sDiscoveryClient(cfg)
```

Comment:
reuse previous client

Reply:
previous client ? only one client created for posthog.

Reply:
done